spel/scripts/amigen8-build.sh

```diff
@@ -0,0 +1,412 @@
+
+   # Set the filesystem-type to use for OS filesystems
+   if [[ ${AMIGENFSTYPE} == "xfs" ]]
+   then
+      err_exit "Using default fstype [xfs] for boot filesysems" NONE
+   fi
+   POSTBUILDCMD+="-f ${AMIGENFSTYPE} "
+
+   # Set location for chroot-env
+   if [[ ${AMIGENCHROOT} == "/mnt/ec2-root" ]]
+   then
+      err_exit "Using default chroot-env location [${AMIGENCHROOT}]" NONE
+   else
+      POSTBUILDCMD+="-m ${AMIGENCHROOT} "
+   fi
+
+   # Set AMI starting time-zone
+   if [[ ${AMIGENTIMEZONE} == "UTC" ]]
+   then
+      err_exit "Using default AMI timezone [${AMIGENCHROOT}]" NONE
+   else
+      POSTBUILDCMD+="-z ${AMIGENTIMEZONE} "
+   fi
+
+   # Return command-string for OS-script
+   echo "${POSTBUILDCMD}"
+}
+
+# Disable strict hostkey checking
+function DisableStrictHostCheck {
+   local HOSTVAL
+
+   if [[ ${1:-} == '' ]]
+   then
+      err_exit "No connect-string passed to function [${0}]"
+   else
+      HOSTVAL="$( sed -e 's/^.*@//' -e 's/:.*$//' <<< "${1}" )"
+   fi
+
+   # Git host-target parameters
+   err_exit "Disabling SSH's strict hostkey checking for ${HOSTVAL}" NONE
+   (
+      printf "Host %s\n" "${HOSTVAL}"
+      printf "  Hostname %s\n" "${HOSTVAL}"
+      printf "  StrictHostKeyChecking off\n"
+   ) >> "${HOME}/.ssh/config" || \
+   err_exit "Failed disabling SSH's strict hostkey checking"
+}
+
+
+
+##########################
+## Main program section ##
+##########################
+
+# Disable strict host-key checking when doing git-over-ssh
+if [[ ${AMIGENSOURCE} =~ "@" ]]
+then
+   DisableStrictHostCheck "${AMIGENSOURCE}"
+fi
+
+# Dismount /oldroot as needed
+if [[ $( mountpoint /oldroot ) =~ "is a mountpoint" ]]
+then
+   err_exit "Dismounting /oldroot..." NONE
+   umount /oldroot || \
+     err_exit "Failed dismounting /oldroot"
+fi
+
+# Null out the build-dev's VTOC
+echo "Checking ${SPEL_AMIGENBUILDDEV} for VTOC to nuke..."
+if [[ $( lsblk -n "${SPEL_AMIGENBUILDDEV}" | tail -1 | cut -d " " -f 1 ) =~ [0-9] ]]
+then
+   dd if=/dev/urandom of="${SPEL_AMIGENBUILDDEV}" bs=1024 count=10240
+
+   echo "Validaing VTOC-state on ${SPEL_AMIGENBUILDDEV}..."
+   if [[ ${SPEL_AMIGENBUILDDEV} =~ /dev/xvd ]]
+   then
+      if [[ $( lsblk -n "${SPEL_AMIGENBUILDDEV}" | tail -1 | cut -d " " -f 1 ) =~ [0-9] ]]
+      then
+         err_exit "Failed clearing VTOC from ${SPEL_AMIGENBUILDDEV}"
+      fi
+   elif [[ ${SPEL_AMIGENBUILDDEV} =~ /dev/nvme ]]
+   then
+      if [[ $( lsblk -n "${SPEL_AMIGENBUILDDEV}" | tail -1 | cut -d " " -f 1 ) =~ p[0-9] ]]
+      then
+         err_exit "Failed clearing VTOC from ${SPEL_AMIGENBUILDDEV}"
+      fi
+   fi
+fi
+
+
+# Ensure build-tools directory exists
+if [[ ! -d ${ELBUILD} ]]
+then
+   err_exit "Creating build-tools directory [${ELBUILD}]..." NONE
+   install -dDm 000755 "${ELBUILD}" || \
+     err_exit "Failed creating build-tools directory"
+fi
+
+# Pull build-tools from git clone-source
+( cd "${ELBUILD}" && git clone --branch "${AMIGENBRANCH}" "${AMIGENSOURCE}" )
```

Comment:
git accepts the directory to clone into. also simplifies the BuildChroot function because the PATHY var becomes unneeded. and the `install` command above becomes unnecessary because git creates the directory.
```suggestion
git clone --branch "${AMIGENBRANCH}" "${AMIGENSOURCE}" "${ELBUILD}"
```

Reply:
Resolved in 44a18c1601eba6f6814467eeb7457c8d8324a09c